schema/table.go

```diff
@@ -80,19 +73,42 @@ type Table struct {
 	// have at least one row.
 	// When IgnoreInTests is true, integration tests won't fetch from this table.
 	// Used when it is hard to create a reproducible environment with a row in this table.
-	IgnoreInTests bool `json:"ignore_in_tests"`
+	IgnoreInTests bool
 
 	// Parent is the parent table in case this table is called via parent table (i.e. relation)
-	Parent *Table `json:"-"`
+	Parent *Table
 
-	PkConstraintName string `json:"pk_constraint_name"`
+	PkConstraintName string
 }
 
 var (
 	reValidTableName  = regexp.MustCompile(`^[a-z_][a-z\d_]*$`)
 	reValidColumnName = regexp.MustCompile(`^[a-z_][a-z\d_]*$`)
 )
 
+// Create a CloudQuery Table abstraction from an arrow schema
+// arrow schema is a low level representation of a table that can be sent
+// over the wire in a cross-language way
+func NewTableFromArrowSchema(sc *arrow.Schema) (*Table, error) {
+	tableMD := sc.Metadata()
+	name, found := tableMD.GetValue(MetadataTableName)
+	if !found {
+		return nil, fmt.Errorf("missing table name")
+	}
+	description, _ := tableMD.GetValue(MetadataTableDescription)
+	fields := sc.Fields()
+	columns := make(ColumnList, len(fields))
+	for i, field := range fields {
+		columns[i] = NewColumnFromArrowField(field)
+	}
+	table := &Table{
+		Name:        name,
+		Description: description,
+		Columns:     columns,
```

Comment:
Couple more metadata items we'll need here I think (e.g. title, pk constraint name, is incremental)

Reply:
Yeah, title is relatively minor and no destinations use it afaik, so maybe no need to include it here. It just depends on whether we're okay with some information being lost in the conversion. Generally speaking it's used by source plugins to generate table docs.